### PR TITLE
Steamswitch webconfig

### PR DESCRIPTION
**Pull request by LoQue90**
Draft for stream switch config on website
for @murmeltier08

### CODE REVIEW

File: src/Storage.cpp [hunk 1]
```diff
@@ -60,6 +60,7 @@ typedef struct __attribute__((packed)) {
     double weightSetpoint;
     double steamkp;
     double steamSetpoint;
+    uint8_t steamSwitchtyp;
```
Comment (kjyv) [inline src/Storage.cpp:63]:
```suggestion
    uint8_t steamSwitchType;
```

File: src/main.cpp [hunk 2]
```diff
@@ -1819,6 +1825,19 @@ void setup() {
...
+    editableVars["STEAM_SWITCHTYP"] = {
+        .displayName = "Steam switch typ",
+        .hasHelpText = true,
+        .helpText = F("0 = switch disabled; 1 = normal switch; 2 = trigger switch"),
+        .type = kUInt8,
+        .section = sTempSection,
+        .position = 28,
+        .show = [] { return true; },
+        .minValue = 0,
+        .maxValue = 1,
```
Comment (kjyv) [inline src/main.cpp:1838]: this should be 2 if 2 is mentioned in the help text

File: src/main.cpp [hunk 3]
```diff
@@ -838,14 +840,18 @@ float filterPressureValue(float input) {
  * @brief steamON & Quickmill
  */
 void checkSteamON() {
-    // check digital GIPO
-    if (digitalRead(PIN_STEAMSWITCH) == HIGH) {
-        steamON = 1;
-    }
+    
+    if (steamSwitchtyp > 0){
+
```
Comment (kjyv) [inline src/main.cpp:845]: unnecessary empty line

File: src/Storage.cpp [hunk 4]
```diff
@@ -107,7 +108,8 @@ static const sto_data_t itemDefaults PROGMEM = {
     "",                                       // STO_ITEM_WIFI_PASSWORD
     SCALE_WEIGHTSETPOINT,                     // STO_ITEM_WEIGHTSETPOINT
     STEAMKP,                                  // STO_ITEM_PID_KP_STEAM
-    STEAMSETPOINT                             // STO_ITEM_STEAM_SETPOINT
+    STEAMSETPOINT,                            // STO_ITEM_STEAM_SETPOINT
+    STEAMSWITCHTYP                            // STO_ITEM_STEAM_SWITCHTYP
```
Comment (kjyv) [inline src/Storage.cpp:112]:
```suggestion
    STEAMSWITCHTYPE                            // STO_ITEM_STEAM_SWITCHTYPE
```
etc.

File: src/main.cpp [hunk 5]
```diff
@@ -225,6 +226,7 @@ SysPara<double> sysParaPreInfPause(&preinfusionpause, PRE_INFUSION_PAUSE_MIN, PR
 SysPara<double> sysParaPidKpSteam(&steamKp, PID_KP_STEAM_MIN, PID_KP_STEAM_MAX, STO_ITEM_PID_KP_STEAM);
 SysPara<double> sysParaSteamSetpoint(&steamSetpoint, STEAM_SETPOINT_MIN, STEAM_SETPOINT_MAX, STO_ITEM_STEAM_SETPOINT);
 SysPara<double> sysParaWeightSetpoint(&weightSetpoint, WEIGHTSETPOINT_MIN, WEIGHTSETPOINT_MAX, STO_ITEM_WEIGHTSETPOINT);
+SysPara<uint8_t> sysParaSteamSwitchtyp(&steamSwitchtyp, 0, 1, STO_ITEM_STEAM_SWITCHTYP);
```
Comment (kjyv) [inline src/main.cpp:229]: max here should also be 2

File: src/defaults.h [hunk 6]
```diff
@@ -69,6 +70,8 @@ int writeSysParamsToStorage(void);
 #define BREW_SETPOINT_MAX 110
 #define STEAM_SETPOINT_MIN 100
 #define STEAM_SETPOINT_MAX 140
+#define STEAM_SWITCHTYP_MIN 0
```
Comment (kjyv) [inline src/defaults.h:73]:
```suggestion
#define STEAM_SWITCHTYPE_MIN 0
```

File: src/defaults.h [hunk 7]
```diff
@@ -69,6 +70,8 @@ int writeSysParamsToStorage(void);
...
+#define STEAM_SWITCHTYP_MAX 2
```
Comment (kjyv) [inline src/defaults.h:74]:
```suggestion
#define STEAM_SWITCHTYPE_MAX 2
```

File: src/main.cpp [hunk 8]
```diff
@@ -225,6 +226,7 @@ SysPara<double> sysParaPreInfPause(&preinfusionpause, PRE_INFUSION_PAUSE_MIN, PR
 SysPara<double> sysParaPidKpSteam(&steamKp, PID_KP_STEAM_MIN, PID_KP_STEAM_MAX, STO_ITEM_PID_KP_STEAM);
 SysPara<double> sysParaSteamSetpoint(&steamSetpoint, STEAM_SETPOINT_MIN, STEAM_SETPOINT_MAX, STO_ITEM_STEAM_SETPOINT);
 SysPara<double> sysParaWeightSetpoint(&weightSetpoint, WEIGHTSETPOINT_MIN, WEIGHTSETPOINT_MAX, STO_ITEM_WEIGHTSETPOINT);
+SysPara<uint8_t> sysParaSteamSwitchType(&steamSwitchType, STEAM_SWITCHTYP_MIN, STEAM_SWITCHTYP_MAX, STO_ITEM_STEAM_SWITCHTYPE);
```
Comment (kjyv) [inline src/main.cpp:229]:
```suggestion
SysPara<uint8_t> sysParaSteamSwitchType(&steamSwitchType, STEAM_SWITCHTYPE_MIN, STEAM_SWITCHTYPE_MAX, STO_ITEM_STEAM_SWITCHTYPE);
```

File: src/main.cpp [hunk 9]
```diff
@@ -1904,6 +1921,11 @@ void setup() {
         pinMode(PIN_POWERSWITCH, INPUT);
     }
 
+    // IF STEAMSWITCH is connected
```
Comment (kjyv) [inline src/main.cpp:1924]:
```suggestion
    // If steam switch is enabled
```

File: src/main.cpp [hunk 10]
```diff
@@ -1904,6 +1921,11 @@ void setup() {
...
+    if (steamSwitchType > 0) {
+        pinMode(PIN_STEAMSWITCH, INPUT);
+    }
+
     // IF Voltage sensor selected
```
Comment (kjyv) [inline src/main.cpp:1929]:
```suggestion
    // If voltage sensor for brew detection is selected
```

File: src/main.cpp [hunk 11]
```diff
@@ -1912,8 +1934,7 @@ void setup() {
         pinMode(PIN_BREWSWITCH, INPUT_PULLDOWN);
     }
 
-    pinMode(PIN_STEAMSWITCH, INPUT_PULLDOWN);
-
+    // IF OLED Display is connected
```
Comment (kjyv) [inline src/main.cpp:1937]:
```suggestion
    // If OLED display is enabled  
```

File: src/main.cpp [hunk 12]
```diff
@@ -1819,6 +1824,19 @@ void setup() {
...
+
+    editableVars["STEAM_SWITCHTYPE"] = {
+        .displayName = "Steam switch type",
+        .hasHelpText = true,
+        .helpText = F("0 = switch disabled; 1 = normal switch; 2 = trigger switch"),
+        .type = kUInt8,
+        .section = sTempSection,
+        .position = 28,
+        .show = [] { return true; },
+        .minValue = 0,
```
Comment (kjyv) [inline src/main.cpp:1836]:
```suggestion
        .minValue = STEAM_SWITCHTYPE_MIN,
```

File: src/main.cpp [hunk 13]
```diff
@@ -1819,6 +1824,19 @@ void setup() {
...
+        .maxValue = 2,
```
Comment (kjyv) [inline src/main.cpp:1837]:
```suggestion
        .maxValue = STEAM_SWITCHTYPE_MAX,
```

**Comment by LoQue90** [comment]
the autofill of vscode screw me up.
i didn´t the the min/max var into the editiablevar because all other vars also have fixed values, thought this is on purpose.

But this still wont work, right now the change on the website will not be stored in the eeprom.
And on top you have to reboot after the change.
also we need a dropdown menue to select the switchtyp

**Comment by LoQue90** [comment]
now part of pr#342

**Pull request closed without merging**